lib/model/Product.dart

```diff
@@ -592,6 +600,12 @@ class Product extends JsonObject {
       }
     }
 
+    if (noNutritionData == true && nutriments?.hasData == true) {
+      throw StateError(
+        'If \'noNutritionData\' is true, no \'nutriments\' should be provided!',
+      );
+    }
+
```

Comment:
I think we would be better off
* making `nutriments` private (therefore `_nutriments`)
* giving "public" access to `nutriments` with getter/setter
* in those getter/setter, include the `noNutritionData` flag, e.g. something like that
```dart
Nutriments? get nutriments => noNutritionData == true ? null : _nutriments
```

Reply:
Your solution is indeed the one I should implement.
However if `nutriments` and/or `nutritionData` is private, there is no way to have a correct mapping with json_serializable… or please correct me on this.

Do you have any idea how to bypass this problem?